File: wms-be/src/main/java/com/restapi/wmsservice/service/BomService.java
package com.restapi.wmsservice.service;

import com.restapi.wmsservice.dto.request.BomRequest;
import com.restapi.wmsservice.dto.response.BomResponse;

import java.util.List;

public interface BomService {
    BomResponse create(BomRequest request);
    List<BomResponse> getAll();
    List<BomResponse> getByParentItemId(Long parentItemId);
    BomResponse getById(Long id);
    BomResponse update(Long id, BomRequest request);
    void delete(Long id);
}

File: wms-fe/src/services/bomService.js
import axiosClient from '../api/axiosClient';

const bomService = {
  getAll: (params) => {
    return axiosClient.get('/boms', { params });
  },
  
  getById: (id) => {
    return axiosClient.get(`/boms/${id}`);
  },
  
  create: (data) => {
    return axiosClient.post('/boms', data);
  },
  
  update: (id, data) => {
    return axiosClient.put(`/boms/${id}`, data);
  },
  
  delete: (id) => {
    return axiosClient.delete(`/boms/${id}`);
  }
};

export default bomService;
